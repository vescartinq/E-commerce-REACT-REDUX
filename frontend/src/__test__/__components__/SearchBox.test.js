import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import SearchBox from '../../components/SearchBox';

const historyMock = { push: jest.fn() };

describe('SearchBox component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchBox history={historyMock} />);
  });

  test('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change input box', () => {
    const input = wrapper.find('input');
    const value = 'Orbea';

    input.simulate('change', { target: { value } });
  });

  test('should call setName and clean input box', () => {
    const value = 'BH';

    // const setName = jest.fn();

    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });

    wrapper.find('form').simulate('submit', { preventDefault() {} });

    // expect(setName).toHaveBeenCalled();
  });
});
