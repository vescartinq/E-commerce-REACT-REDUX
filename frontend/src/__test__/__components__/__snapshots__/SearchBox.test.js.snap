// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component should render component 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <input
      id="q"
      name="q"
      onChange={[Function]}
      type="text"
    />
    <button
      className="primary"
      type="submit"
    >
      <i
        className="fa fa-search"
      />
    </button>
  </div>
</form>
`;
